app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1282,17 +1295,23 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 
             // Create a new instance of DatePickerDialog and return it
             DatePickerDialog datePickerDialog = new DatePickerDialog(getActivity(), this, year, month, day);
-            datePickerDialog.getDatePicker().setMinDate(getMinDateOfDatePicker());
+            datePickerDialog.getDatePicker().setMinDate(minDate != null ? minDate.getTime() : getDefaultMinDateOfDatePicker());
+            datePickerDialog.getDatePicker().setMaxDate(maxDate != null ? maxDate.getTime() : getDefaultMaxDateOfDatePicker());
```

Comment:
Technically I don't need to set the default max date here, but having an if statement here didn't look that neat to me 😄 :
```
if (maxDate != null) {
    datePickerDialog.getDatePicker().setMaxDate(maxDate.getTime());
}
```
Let me know if I should use the if statement instead :)